app/scripts/controllers/edit/route.js

```diff
@@ -73,16 +88,13 @@ angular.module('openshiftConsole')
 
         // Update the fields in the route from what was entered in the form.
         var updateRouteFields = function() {
-          var serviceName = _.get($scope, 'routing.service.metadata.name');
+          var serviceName = _.get($scope, 'routing.to.service.metadata.name');
           _.set(route, 'spec.to.name', serviceName);
-
-          // Remove the host.generated annotation if the host was edited.
-          if (_.get(route, ['metadata', 'annotations', 'openshift.io/host.generated']) === 'true' &&
-              _.get(route, 'spec.host') !== $scope.routing.host) {
-            delete route.metadata.annotations['openshift.io/host.generated'];
+          var weight = _.get($scope, 'routing.to.weight');
+          if (!isNaN(weight)) {
```

Comment:
are you not doing number validation on the inputs, or is this just an extra sanity check